components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt

```diff
@@ -113,7 +113,16 @@ internal class PingMaker(
      */
     fun collect(storage: String): String {
         val jsonPing = storageManager.collect(storage)
+
+        // TODO, maybe need to not add the 'ping_info' to the jsonPing if the jsonPing is already empty
+        // This could then be used by consuming functions (i.e. sendPing()) to selectively send the
+        // ping or not, depending on whether it was empty or not.
+        if (jsonPing.length() == 0) {
```

Comment:
Instead of relying on special values, we could make this function return a nullable String (i.e. `String?`). This way we could simply null-check at the call site and return null here.

Reply:
Sure, nullable is good and can make the calling function more Kotlin-like.

Reply:
We should probably add a comment to `StorageEngineManager.collect` that it should return a zero-length `JSONObject` whenever there is no actual content.  And I'm not sure it actually currently works that way.  (See below).